Rules/CTF/Scripts/CTF_Banners.as

```diff
@@ -0,0 +1,135 @@
+#define CLIENT_ONLY;
+
+#include "GameStateBanners.as";
+
+Banner@ getBuildBanner()
+{
+	u32 duration = 5 * getTicksASecond();
+	string text = "Build defenses!";
+	string secondary_text = "Increased build speed and resupplies";
+	int team = 255; // not needed
+
+	Icon@ left_icon = Icon("InteractionIcons.png", 21, Vec2f(32, 32), Vec2f(160, 32), 0);
+	Icon@ right_icon = Icon("InteractionIcons.png", 21, Vec2f(32, 32), Vec2f(96, -32), 0);
+
+	Banner banner(duration, text, left_icon, right_icon, team, false, null, true, secondary_text);
+
+	return banner;
+}
+
+Banner@ getGameBanner(int team=0)
+{
+	u32 duration = 5 * getTicksASecond();
+	string text = "Capture the flag!";
+
+	Icon@ left_icon = Icon("BannerIcons.png", 0, Vec2f(32, 32), Vec2f(184, 32), team);
+	Icon@ right_icon = Icon("BannerIcons.png", 1, Vec2f(32, 32), Vec2f(120, -32), team);
+
+	Banner banner(duration, text, left_icon, right_icon);
+
+	return banner;
+}
+
+Banner@ getWinBanner(int team=0)
+{
+	u32 duration = 8 * getTicksASecond();
+	string text = "{TEAM} team wins";
+	string teamName = (team == 0 ? "Blue" : "Red");
+	string actual_text = teamName + " team wins";
+
+	Icon@ team_icon = getTeamIcon(team);
+	Icon@ left_icon = Icon("MenuItems.png", 31, Vec2f(32, 32), Vec2f(192, 32), team);
+	Icon@ right_icon = Icon("MenuItems.png", 31, Vec2f(32, 32), Vec2f(128, -32), team);
+
+	Banner banner(duration, text, left_icon, right_icon, team, true, team_icon);
+
+	return banner;
+}
+
+Banner@[] banners;
+
+enum BannerType
+{
+	WARMUP_START = 0,
+	GAME_START,
+	GAME_END
+};
+
+void onInit(CRules@ this)
+{
+	banners.insertAt(WARMUP_START, getBuildBanner());
+	banners.insertAt(GAME_START, getGameBanner());
+	banners.insertAt(GAME_END, getWinBanner());
+}
+
+void onReload(CRules@ this)
+{
+	banners.clear();
+	onInit(this);
+}
+
+void onRender(CRules@ this)
```

Comment:
pretty much everything here can be `onTick`, just change line 75 to `!this.get_bool("Draw Banner")` and move the draw code and property setter to be the only thing in `onRender`

Reply:
Moved some stuff to onTick